server/controllers/favorite.js

```diff
@@ -0,0 +1,51 @@
+import { Op } from "sequelize";
+import models from "../models";
+import { httpResponse } from "../helpers/http";
+
+const { Favourites } = models;
+/**
+ * @description This asynchronous function favorite an article
+ *          that is not yet favorited and increase the favorite count on the article
+ *          and also remove favorite on an article that is already
+ * favorited and reduce the favorite count.s
+ * @param  {httpRequest} req
+ * @param  {httpResponse} res
+ * @methods favoriteorUnFavoriteArticle
+ * @returns {httpResponse} res
+ */
+export const favoriteOrUnFavoriteArticle = async (req, res) => {
+  const {
+    article: { slug, id },
+    user: { userId },
+    article
+  } = req;
+
+  const favourites = await Favourites.findOne({
+    where: {
+      articleSlug: {
+        [Op.eq]: slug
+      },
+      userId: {
+        [Op.eq]: userId
+      }
+    }
+  });
+
+  if (!favourites) {
+    await Favourites.create({
+      userId,
+      articleSlug: slug,
+      articleId: id
+    });
+    await article.increaseFavoriteCount();
```

Comment:
You don't have to re-event the wheel, Sequelize has a function that does this same thing.
this is all you need here: await article.increment('favouriteCount');